src/Meta.php

```diff
@@ -0,0 +1,203 @@
+		}
+		return $description;
+	}
+
+	/**
+	 * Get text trimmed for a meta description.
+	 *
+	 * @param string $text The text to trim.
+	 * @return string
+	 */
+	public static function get_trimmed_meta_description( $text ) {
+		$limit = 160;
+
+		if ( strlen( $text ) <= $limit ) {
+			return $text;
+		}
+
+		$wrapped_text = explode( '\n', wordwrap( $text , $limit, '\n' ) );
+
+		return $wrapped_text[0];
+	}
+
+	/**
+	 * Get the post's og title.
+	 *
+	 * @param \WP_Post $post The post.
+	 * @return string
+	 */
+	public static function get_post_og_title( $post ) {
+		$title = get_post_meta( $post->ID, '_yoast_wpseo_opengraph-title', true );
+
+		if ( empty( $title ) ) {
+			$title = self::get_post_meta_title( $post );
+		}
+
+		return $title;
+	}
+
+	/**
+	 * Get the post's og description.
+	 *
+	 * @param \WP_Post $post The post.
+	 * @return string
+	 */
+	public static function get_post_og_description( $post ) {
+		$description = get_post_meta( $post->ID, '_yoast_wpseo_opengraph-description', true );
+
+		if ( empty( $description ) ) {
+			$description = self::get_post_meta_description( $post );
+		}
+
+		return $description;
+	}
+
+	/**
+	 * Get the post's og image.
+	 *
+	 * @param \WP_Post $post The post.
+	 * @return string
+	 */
+	public static function get_post_og_image( $post ) {
+		$image = get_post_meta( $post->ID, '_yoast_wpseo_opengraph-image', true );
+
+		if ( empty( $image ) ) {
+			$image = self::get_fallback_image( $post );
+		}
+
+		return $image;
+	}
+
+	/**
+	 * Get the post's twitter title.
+	 *
+	 * @param \WP_Post $post The post.
+	 * @return string
+	 */
+	public static function get_post_twitter_title( $post ) {
+		$title = get_post_meta( $post->ID, '_yoast_wpseo_twitter-title', true );
+
+		if ( empty( $title ) ) {
+			$title = self::get_post_meta_title( $post );
+		}
+
+		return $title;
+	}
+
+	/**
+	 * Get the post's twitter description.
+	 *
+	 * @param \WP_Post $post The post.
+	 * @return string
+	 */
+	public static function get_post_twitter_description( $post ) {
+		$description = get_post_meta( $post->ID, '_yoast_wpseo_twitter-description', true );
+
+		if ( empty( $description ) ) {
+			$description = self::get_post_meta_description( $post );
+		}
+
+		return $description;
+	}
+
+	/**
+	 * Get the post's twitter image.
+	 *
+	 * @param \WP_Post $post The post.
+	 * @return string
+	 */
+	public static function get_post_twitter_image( $post ) {
+		$image = get_post_meta( $post->ID, '_yoast_wpseo_twitter-image', true );
+
+		if ( empty( $image ) ) {
+			$image = self::get_fallback_image( $post );
+		}
+
+		return $image;
+	}
+
+	/**
+	 * Get a fallback image for the post.
+	 *
+	 * @param \WP_Post $post The post.
+	 * @return mixed
+	 */
+	private static function get_fallback_image( $post ) {
+		$image = wp_get_attachment_url( get_post_thumbnail_id( $post->ID ) );
+
+		if ( empty( $image ) ) {
+			$logo = Acf::get_option_field( 'site_logo' );
```

Comment:
We would need to add wp-elements as dependency for this, as the site_logo comes from there. I guess we can do it. I was thinking we could even create wrapper functions in wp-elements for important data, eg `get_site_logo()`